### PR TITLE
limit max screen title length to window width

### PR DESCRIPTION
**Pull request by ryenus**
Applies screen title truncating to all InfoScreen classes.

@cgzones Possible to take a look? Thank you!

### CODE REVIEW

File: InfoScreen.c [hunk 1]
```diff
@@ -42,14 +42,19 @@ InfoScreen* InfoScreen_done(InfoScreen* this) {
 void InfoScreen_drawTitled(InfoScreen* this, const char* fmt, ...) {
    va_list ap;
    va_start(ap, fmt);
+
+   char* title = xMalloc(COLS + 1);
+   int len = vsnprintf(title, COLS + 1, fmt, ap);
+   if (len > COLS) memset(&title[COLS - 3], '.', 3);
+
```
Comment (BenBE) [inline InfoScreen.c:49]: Could we try to move this allocation into the lifetime of the window?

File: InfoScreen.c [hunk 2]
```diff
@@ -42,14 +42,19 @@ InfoScreen* InfoScreen_done(InfoScreen* this) {
 void InfoScreen_drawTitled(InfoScreen* this, const char* fmt, ...) {
    va_list ap;
    va_start(ap, fmt);
+
+   char* title = xMalloc(COLS + 1);
+   int len = vsnprintf(title, COLS + 1, fmt, ap);
+   if (len > COLS) memset(&title[COLS - 3], '.', 3);
```
Comment (BenBE) [inline InfoScreen.c:48]: Please avoid braceless if-statements.

File: InfoScreen.c (hunk 1 above)
Comment (cgzones) [reply InfoScreen.c:49]: I think this is not needed:
`InfoScreen_drawTitled` is called in the `_draw` callbacks of the Screen classes.
The `_draw` callback is called (besides the initial one) on window-resize, F5 key and Crtl-L key.
These events are rather rare, so caching seems not worth it and on resize we would to re-compute anyway.

File: InfoScreen.c (hunk 2 above)
Comment (cgzones) [reply InfoScreen.c:48]: I am not sure if we have agreed on a brace-style for one-line branches. @natoscott @ginggs @fasterit 
But at least put the branch into a separate line.

Comment (BenBE) [reply InfoScreen.c:48]: The original source had it on multiple lines. ;-)

Comment (cgzones) [reply InfoScreen.c:48]: Good point, another topic for a future style guide.

File: InfoScreen.c (hunk 1 above)
Comment (BenBE) [reply InfoScreen.c:49]: Okay. Sounds reasonable.

File: InfoScreen.c (hunk 2 above)
Comment (ryenus) [reply InfoScreen.c]: OK, braces added, though I'm fine with both, but having an agreed style guide is great. 👍

**Comment by cgzones** [comment]
Applied, thanks.

**Pull request merged**